src/pydap/model.py

```diff
@@ -512,32 +532,76 @@ def _get_data(self):
 
     data = property(_get_data, _set_data)
 
-    def __len__(self):
-        return len(self.data)
-
-    def __iter__(self):
+    def iterdata(self):
         for line in self.data:
             yield tuple(map(decode_np_strings, line))
 
+    _iter_deprecation_msg = ('Iteration now yields children names. '
+                             'This means that in the future '
+                             '``for val in sequence: ...`` '
+                             'will give children names. '
+                             'To iterate over data the construct '
+                             '``for val in sequence.iterdata(): ...``'
+                             'is available and will be supported in the'
+                             'future')
+
+    def __iter__(self):
+        # This method should be removed once the deprecation is
```

Comment:
For all of these future deprecation comments, I think that it should be specified in the comment what the target for deprecation is. If somehow these get overlooked during the course of another couple releases, it would be nice to have it say exactly what our intent was, so that we can compare that to the future present. :) E.g. if the comment says "remove this by version 3.4", and we're on version 3.5, it's obvious to even the casual observer that action is required.